aptos-move/e2e-tests/src/executor.rs

```diff
@@ -1010,34 +1025,38 @@ impl FakeExecutor {
         self.exec_module(&Self::module(module_name), function_name, type_params, args)
     }
 
-    pub fn try_exec_entry_with_features(
+    pub fn try_exec_entry_with_resolver(
         &mut self,
         senders: Vec<AccountAddress>,
         entry_fn: &EntryFunction,
-        features: &Features,
+        resolver: &impl AptosMoveResolver,
+        v2_flag: bool,
     ) -> Result<(WriteSet, Vec<ContractEvent>), VMStatus> {
-        let resolver = self.data_store.as_move_resolver();
-
         let timed_features = TimedFeaturesBuilder::enable_all()
             .with_override_profile(TimedFeatureOverride::Testing)
             .build();
 
+        let mut features = Features::fetch_config(resolver).unwrap_or_default();
+        if v2_flag {
+            features.enable(FeatureFlag::VM_BINARY_FORMAT_V7);
```

Comment:
Can we remove this ad-hoc logic? Or can the caller of this function use the features args to set the flag?

Reply:
done